oauth/lib/redirecturiparser.cpp

```diff
@@ -0,0 +1,51 @@
+/****************************************************************************
+**
+** Copyright (C) 2021 Open Mobile Platform LLC
+**
+****************************************************************************/
+
+#include "redirecturiparser_p.h"
+
+#include <QtCore/QUrl>
+#include <QtCore/QUrlQuery>
+#include <QtCore/QStringList>
+#include <QtCore/QString>
+#include <QtCore/QByteArray>
+#include <QtCore/QPair>
+#include <QtCore/QList>
+
+QVariantMap SailfishOS::OAuth::parseRedirectUri(const QString &redirectUri, QString *errorMessage)
+{
+    QVariantMap data;
+
+    const QUrl url(redirectUri);
+    const QUrlQuery query(url.query());
+    const QString fragment = url.fragment(QUrl::FullyEncoded);
```

Comment:
Where were it used? I don't find this place

Reply:
In the line below, but not necessary to be separated, so I have combined those two lines now.  Done.